### PR TITLE
fix: ID param for DELETE ssh_tunnel endpoint

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
The DELETE `database/{id}/ssh_tunnel `endpoint was incorrectly expecting an SSH Tunnel ID rather than the Database ID. This PR fixes it.

It deprecates the endpoint as the ssh_tunnel should be a property of the database and should be updated according to the database update endpoint.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N.A.

### TESTING INSTRUCTIONS
- [ ] CI should pass

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/databases/api.py [hunk 1]
```diff
@@ -1447,6 +1447,7 @@ def validate_parameters(self) -> FlaskResponse:
     @expose("/<int:pk>/ssh_tunnel/", methods=("DELETE",))
     @protect()
     @statsd_metrics
+    @deprecated(deprecated_in="5.0")
```
Comment (michael-s-molina) [inline superset/databases/api.py:1450]:
```suggestion
    @deprecated(deprecated_in="4.0")
```

**Review by michael-s-molina — APPROVED** [review]
LGTM

File: superset/databases/api.py [hunk 2]
```diff
@@ -1447,6 +1447,7 @@ def validate_parameters(self) -> FlaskResponse:
     @expose("/<int:pk>/ssh_tunnel/", methods=("DELETE",))
     @protect()
     @statsd_metrics
+    @deprecated(deprecated_in="4.0")
```
Comment (eschutho) [inline superset/databases/api.py:1450]: @geido doesn't this mean that we'll be removing this endpoint in 4.0?

Comment (michael-s-molina) [reply superset/databases/api.py:1450]: No. It means that the endpoint was deprecated in 4.0 to be removed in a future version. Possibly 5.0.

Comment (geido) [reply superset/databases/api.py:1450]: Yeah the idea is indeed to remove it in a future version

**Pull request merged**